build/Dockerfile

```diff
@@ -1,46 +1,289 @@
+	yum install -y ca-certificates nginx-plus-${NGINX_PLUS_VERSION}
+
+
+############################################# Build image for Opentracing #############################################
+
+FROM debian as opentracing-builder
+
+ARG NGINX_VERSION
+ARG NGINX_OPENTRACING=0.10.0
+ARG OPENTRACING_VERSION=1.6.0
+
+RUN apt-get update && apt-get install -y -q --fix-missing --no-install-recommends \
+	autoconf \
+	automake \
+	build-essential \
+	cmake \
+	git \
+	libcurl4-openssl-dev \
+	libcurl4-openssl-dev \
+	libgeoip-dev \
+	liblmdb-dev \
+	libpcre3-dev \
+	libprotobuf-dev \
+	libssl-dev \
+	libtool \
+	libxml2-dev \
+	libyajl-dev \
+	pkgconf \
+	protobuf-compiler \
+	zlib1g-dev
+
+# Get nginx to build against
+RUN curl -sS -O -L http://nginx.org/download/nginx-${NGINX_VERSION}.tar.gz && \
+	tar zxvf nginx-${NGINX_VERSION}.tar.gz && rm -f nginx-${NGINX_VERSION}.tar.gz
+
+RUN git clone --branch v${NGINX_OPENTRACING} https://github.com/opentracing-contrib/nginx-opentracing.git
+RUN git clone --branch v${OPENTRACING_VERSION} https://github.com/opentracing/opentracing-cpp.git
+
+RUN mkdir -p opentracing-cpp/.build && cd opentracing-cpp/.build && \
+	cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DBUILD_STATIC_LIBS=OFF -DBUILD_MOCKTRACER=OFF -DENABLE_LINTING=OFF .. && \
+	make && make install
+
+RUN cd nginx-${NGINX_VERSION} && \
+	./configure \
+	--with-compat --with-debug \
+	--add-dynamic-module=/nginx-opentracing/opentracing && \
+	make modules
+
+FROM debian AS tracer-downloader
+
+ARG JAEGER_VERSION=v0.4.2
+
+RUN set -x \
+	&& apt-get update \
+	&& apt-get install --no-install-recommends --no-install-suggests -y ca-certificates apt-transport-https wget \
+	&& wget https://github.com/jaegertracing/jaeger-client-cpp/releases/download/${JAEGER_VERSION}/libjaegertracing_plugin.linux_amd64.so -O /usr/local/lib/libjaegertracing_plugin.so
+
+
+FROM debian as opentracing
+ARG NGINX_VERSION
+ARG OPENTRACING_VERSION=1.6.0
+
+COPY --from=opentracing-builder /nginx-${NGINX_VERSION}/objs/ngx_http_opentracing_module.so /usr/lib/nginx/modules/ngx_http_opentracing_module.so
+COPY --from=opentracing-builder /usr/local/lib/libopentracing.so.${OPENTRACING_VERSION} /usr/local/lib/libopentracing.so.1
+COPY --from=tracer-downloader /usr/local/lib/libjaegertracing_plugin.so /usr/local/lib/libjaegertracing_plugin.so
+
+RUN ldconfig
+
+############################################# Build image for Opentracing with NGINX Plus #############################################
+FROM debian-plus as opentracing-plus
+
+ENV NGINX_OPENTRACING_MODULE_VERSION 23+0.9.0-1~buster
+
+RUN --mount=type=secret,id=nginx-repo.crt,dst=/etc/ssl/nginx/nginx-repo.crt,mode=0644 \
+	--mount=type=secret,id=nginx-repo.key,dst=/etc/ssl/nginx/nginx-repo.key,mode=0644 \
+	apt-get update && \
+	apt-get install --no-install-recommends --no-install-suggests nginx-plus-module-opentracing=${NGINX_OPENTRACING_MODULE_VERSION} && \
+	rm -rf /var/lib/apt/lists/*
```

Comment:
what about removing files in apt.conf.d like we do here https://github.com/nginxinc/kubernetes-ingress/blob/master/build/DockerfileWithOpentracingForPlus#L59 ?